src/Grid/grid.jl

```diff
@@ -65,6 +75,20 @@ corresponds to the local index into this tuple.
 """
 vertices(::AbstractCell)
 
+"""
+    reference_edges(::AbstractRefShape)
+    reference_edges(::AbstractCell)
+
+Returns a tuple of 2-tuples containing the ordered local node indices corresponding to
+the vertices that define an *oriented edge*.
+
+An *oriented edge* is an edge with the first node having the lowest local index and the other
+node the secon dindex.
+
+Note that the vertices are sufficient to define a face uniquely.
```

Comment:
```suggestion
Returns a tuple of 2-tuples containing the local node indices corresponding to
the vertices that define the edge. The ordering determines the positive direction of this edge.
```

Reply:
```suggestion
Returns a tuple of 2-tuples containing the ordered local node indices (corresponding to the vertices) that define an edge.

Note that the vertices are sufficient to define an edge uniquely.
```

Reply:
I think the phrase "up to its orientation" isn't so clear (could be since I'm not a native English speaker) (does it include the orientation or not?)

In general I'm not sure about the last sentence, since without the global node indices, the edge isn't uniquely defined? So doesn't this statement rather belong to `edges` and not `reference_edges`? 

I think we should write that `edges(c::AbstractCell) returns the global node indices corresponding to the local node indices of c's reference shape's vertices`, and in `edges` state that `The global node indices of the vertices are sufficient to determine both the edge and its direction uniquely`.

Reply:
> Do we have some reference in the docs what positive/negative orientation means in this context?

No, since that PR isn't merged I don't think so. But sure, the definition is the same, and my logic to how this should work in Ferrite (with #798) is that 
* The dof-distriubtion takes care of the "dof location" for each base functions. E.g. if the edge is reversed, we also need to reverse the positions (this seemed to work out of the box which was nice, but I guess expected since this is the same problem for standard nodal interoplations). 
* The `reinit!` takes care of flipping the sign of the vectorial base functions if the edge is reversed. Similarly for faces, we need to determine if the face is positive or negative, and flip the sign accordingly. But the rotation doesn't matter for `reinit!`, as the location has already been taken care of by the dof-distribution. 


Another side-note, now we should rename this to `EdgeOrientationInfo` after redefining edges :)

Reply:
> Maybe the missing information here is that my wording is "oriented" for the "not sorted" entities as they are. Hence the reference edge would always be oriented, but not necessarily unique (and we also cannot determine this with the information at hand). The physical edge (queried via edges would then still be oriented and we can query details on its orientation, which allows us to define some new entity, which is still an edge through sortedge. I am sure that we are missing this important piece of information in the doc strings, which is why I wanted to add it.

As I wrote above, I think "ordered" => "not sorted" and "sorted" => "unique entity description" is clear. I would rather favor adding this description to `edges` and `faces`, rather than `reference_edges` and `reference_faces`, since that is where we have this information. Then it should be explained that `reference_*` provide the local vertices that are used to obtain this information from cells sharing the same reference shape.

Reply:
Okay, I put a bit more time into this and my proposal is as follows. In this PR we do
```suggestion
Returns a tuple of 2-tuples containing the ordered local node indices (corresponding to the vertices) that define an edge.
```
and, in a separate PR, we add a subsection on the orientation stuff into the Grid topic guide together with renaming SurfaceOrientation and PathOrientationInfo to FaceOrientationInfo and EdgeOrientationInfo. The orientation API should then be made be public, but marked experimental for now, as users could use this stuff for custom dof handlers, custom elements and custom processing routines.

Reply:
Great! 

> In this PR we do

The above suggested change for `reference_edges` is good. 

Could do the equivalent change `reference_faces` here. 
For `reference_facets`, I guess we should just refer to `reference_edges` and `reference_faces`? Something like 
```
Returns a tuple of tuples which define a facet, see `reference_edges` and `reference_faces` for further details.
``` 

> The orientation API should then be made be public

Not sure. I thought `SurfaceOrientation` isn't properly working yet? What's the advantage of making public but experimental over simply keeping in devdocs? But anyways, let's discuss that in the relevant (at minimum, renaming) pr!

Reply:
I can implement SurfaceOrientation if you need it. This is the easy part, no worries. The "hard" part is high order 3D ansatz functions, because you need to think about the permutation map for all combinations of admissible surface orientations.

Reply:
I don't need it now at least, I also don't see a need to rush this to public api as long as it is documented in devdocs.